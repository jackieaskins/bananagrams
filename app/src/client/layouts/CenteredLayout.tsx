import { Center, Container } from "@chakra-ui/react";

type CenteredLayoutProps = {
  children: React.ReactNode;
};

export default function CenteredLayout({
  children,
}: CenteredLayoutProps): JSX.Element {
  return (
    <Center height="100vh">
      <Container>{children}</Container>
    </Center>
  );
}
