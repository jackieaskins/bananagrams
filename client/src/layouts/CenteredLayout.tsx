import { Grid, GridSize } from '@material-ui/core';

type CenteredLayoutProps = {
  children: React.ReactNode;
  width?: GridSize;
};

const CenteredLayout = ({
  children,
  width = 6,
}: CenteredLayoutProps): JSX.Element => (
  <Grid container justify="center" alignContent="center">
    <Grid item xs={width}>
      {children}
    </Grid>
  </Grid>
);

export default CenteredLayout;
