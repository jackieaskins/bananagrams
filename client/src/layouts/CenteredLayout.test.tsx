import { shallow } from 'enzyme';

import CenteredLayout from './CenteredLayout';

describe('<CeneteredLayout />', () => {
  it('renders properly with default width', () => {
    expect(
      shallow(<CenteredLayout>Children</CenteredLayout>)
    ).toMatchSnapshot();
  });

  it('renders properly with passed in width', () => {
    expect(
      shallow(<CenteredLayout width={2}>Children</CenteredLayout>)
    ).toMatchSnapshot();
  });
});
