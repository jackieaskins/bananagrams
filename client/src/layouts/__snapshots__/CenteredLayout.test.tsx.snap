// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CeneteredLayout /> renders properly with default width 1`] = `
<WithStyles(ForwardRef(Grid))
  alignContent="center"
  container={true}
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={6}
  >
    Children
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;

exports[`<CeneteredLayout /> renders properly with passed in width 1`] = `
<WithStyles(ForwardRef(Grid))
  alignContent="center"
  container={true}
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
    xs={2}
  >
    Children
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
